packages/veritone-react-common/src/components/FaceEngineOutput/index.js

```diff
@@ -0,0 +1,490 @@
+      if (entity && entity.name && this.props.libraries.length) {
+        let library = find(this.props.libraries, { id: entity.libraryId });
+        let face = {
+          entityId: faceObj.object.entityId,
+          libraryId: library.id,
+          libraryName: library.name,
+          fullName: entity.name,
+          entity: {
+            ...entity,
+            libraryId: library.id,
+            libraryName: library.name
+          },
+          profileImage: entity.profileImageUrl,
+          count: 1,
+          timeSlots: [
+            {
+              stopTimeMs: faceObj.stopTimeMs,
+              startTimeMs: faceObj.startTimeMs,
+              originalImage: faceObj.object.uri,
+              confidence: faceObj.object.confidence
+            }
+          ],
+          stopTimeMs: faceObj.stopTimeMs
+        };
+        let flag = true;
+        recognizedEntityObjects.forEach(function checkDuplicate(
+          elem,
+          index,
+          array
+        ) {
+          if (elem.entityId === face.entityId) {
+            elem.count++;
+            let timeStamp = {
+              stopTimeMs: faceObj.stopTimeMs,
+              startTimeMs: faceObj.startTimeMs,
+              originalImage: faceObj.object.uri,
+              confidence: faceObj.object.confidence
+            };
+            elem.timeSlots.push(timeStamp);
+            elem.stopTimeMs =
+              elem.stopTimeMs <= faceObj.stopTimeMs
+                ? faceObj.stopTimeMs
+                : elem.stopTimeMs;
+            flag = false;
+          }
+        });
+        if (flag) {
+          recognizedEntityObjectMap[face.entityId] = face;
+          recognizedEntityObjects.push(face);
+        }
+
+        let matchNamespace = this.getFrameNamespaceForMatch(faceObj);
+        if (matchNamespace) {
+          let secondSpots = this.getArrayOfSecondSpots(faceObj);
+          secondSpots.forEach(second => {
+            if (!secondMap[second]) {
+              secondMap[second] = {};
+            }
+            if (!secondMap[second][matchNamespace]) {
+              secondMap[second][matchNamespace] = {
+                startTimeMs: faceObj.startTimeMs,
+                stopTimeMs: faceObj.stopTimeMs,
+                originalImage: faceObj.object.uri,
+                entities: [],
+                boundingPoly: faceObj.object.boundingPoly
+              };
+            }
+
+            let match = {
+              confidence: faceObj.object.confidence,
+              entityId: faceObj.object.entityId
+            };
+
+            secondMap[second][matchNamespace].entities.push(match);
+
+            secondMap[second][matchNamespace].entities.sort((a, b) => {
+              return b.confidence - a.confidence;
+            });
+          });
+        }
+      } else if (!faceObj.entityId) {
+        detectedFaceObjects.push(faceObj);
+      }
+    });
+
+    // Loop through faces and namespace by library
+    let entitiesByLibrary = {};
```

Comment:
no need to iterate over this again, you can include it as part of the initial iteration. See my comment above for more details.